enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,269 @@
+clusters can update to this, the ISO image will be included in the release
+payload. This will be balanced out by removing the RHEL image currently shipped
+in the release payload.
+
+## Motivation
+
+Currently, the deploy disk image (i.e. the image running IPA -
+`ironic-python-agent`) is a RHEL kernel plus initrd that is installed (from an
+RPM) into the `ironic-ipa-downloader` container image, which in turn is part of
+the OpenShift release payload. When the metal3 Pod starts up, the disk image is
+copied from the container to a HostPath volume whence it is available to
+Ironic.
+
+The target OS disk image is a separate CoreOS QCOW2 image. The URL for this is
+known by the installer. It points to the public Internet by default and may be
+customised by the user to allow disconnected installs. The URL is stored in the
+Provisioning CR at install time and never updated automatically. The image
+itself is downloaded once and permanently cached on all of the master nodes.
+Never updating the image is tolerable because, upon booting, the CoreOS image
+will update itself to the version matching the cluster it is to join. It
+remains suboptimal because new Machines will take longer and longer (and more
+and more bandwidth) to join as the cluster ages. This issue exists on all
+platforms, and is the subject of a [long-standing enhancement
+proposal](https://github.com/openshift/enhancements/pull/201). Other issues
+specific to the baremetal platform are that boot times for bare metal servers
+can be very long (and therefore the reboot is costly), and that support for
+particular hardware may theoretically require a particular version of CoreOS.
+
+We are changing the deploy disk image to use the same CoreOS images used for
+UPI deployments. These take the form of both a live ISO (for hosts that can use
+virtualmedia) and of a kernel + initrd + rootfs (for hosts that use PXE). When
+upgrading an existing disconnected cluster, we currently have no way to acquire
+these images without the user manually intervening to mirror them.
+
+Like the QCOW2 provisioning disk image, the URLs for these images are known by
+the installer, but they point to the cloud by default and would have to be
+customised by the user at install time to allow disconnected installs.
+Following a similar approach to that currently used with the QCOW2 also
+effectively extends the limitation that we are not updating the provisioning OS
+image to include the deploy image as well.
+
+The agent itself (IPA) is delivered separately, in a container image as part of
+the OpenShift release payload, so in any event we will continue to be able to
+update IPA.
+
+We wish to solve the problems with obtaining an up-to-date CoreOS by including
+it in the release payload.
+
+### Goals
+
+* Ensure that no matter which version of OpenShift a cluster was installed
+  with, we are able to deliver updates to IPA and the OS it runs on.
```

Comment:
Hmm, if you avoid bumping this image with each release and as a result keep the reboot (both stated non-goals), it was initially unclear to me what the win was from this entry.  Going over the *Motivation* and *User Stories* sections again, I think it might be worth explicitly calling out these sub-cases here:
 
* More convenient experience for folks on restricted networks, who would no longer need to mirror the boot image separately, since it would be mirrored with the rest of the OCP release.
* Support for adding nodes with newer hardware that was not supported by the boot image mirrored when the cluster was installed.
* Avoiding a window of pre-pivot bug/security exposure by sun-setting older boot images.